libarchive/archive_write_disk_posix.c

```diff
@@ -1648,6 +1638,17 @@ _archive_write_disk_finish_entry(struct archive *_a)
 	}
 
 	/*
+	 * Security-related extended attributes (such as
+	 * security.capability or security.ima on Linux) have to be restored last,
+	 * since they're implicitly removed by other file changes like setting
+	 * times.
+	 */
+	if (a->todo & TODO_XATTR) {
+		int r2 = set_xattrs(a);
+		if (r2 < ret) ret = r2;
+	}
+
+	/*
```

Comment:
Not really. This particular case I didn't test and it's a valid concern.
Actually this PR can be safely closed now because the root cause has been fixed in the kernel's patch queue [1].

[1] https://git.kernel.org/cgit/linux/kernel/git/zohar/linux-integrity.git/commit/?h=next-4.7&id=c737f74525784a4e8af3fcfae480fcb31e4fd65a